gsy_framework/data_classes.py

```diff
@@ -368,15 +303,24 @@ class TradeBidOfferInfo:
     trade_rate: Optional[float]
     matching_requirements: Optional[Dict] = None
 
+    def serializable_dict(self) -> Dict:
+        """Return a json serializable representation of the class."""
+        return asdict(self)
+
     def to_json_string(self) -> str:
         """Return json string of the representation."""
-        return json.dumps(asdict(self), default=json_datetime_serializer)
+        return json.dumps(self.serializable_dict(), default=json_datetime_serializer)
 
     @staticmethod
-    def from_json(trade_bid_offer_info: str) -> "TradeBidOfferInfo":
+    def from_json(trade_bid_offer_info: Union[str, Dict]) -> "TradeBidOfferInfo":
         """Return TradeBidOfferInfo object from json representation."""
-        info_dict = json.loads(trade_bid_offer_info)
-        return TradeBidOfferInfo(**info_dict)
+        if isinstance(trade_bid_offer_info, str):
+            info_dict = json.loads(trade_bid_offer_info)
+            return TradeBidOfferInfo(**info_dict)
+        return TradeBidOfferInfo(**trade_bid_offer_info)
+
+    def __eq__(self, other: "TradeBidOfferInfo") -> bool:
+        return self.to_json_string() == other.to_json_string()
```

Comment:
I think that it is an overkill because in order for this occurrence to happen, we have to dynamically modify the class `__dict__` by monkeypatching, which I am fairly certain we will not do here. I'll fix it though, just for completeness purpose.